engines/python/setup/djl_python/tensorrt_llm.py

```diff
@@ -58,56 +33,24 @@ def __init__(self):
-        # If option.model_id points to a s3 bucket, we download it and set model_id to the download directory.
-        # Otherwise we assume model artifacts are in the model_dir
         model_id_or_path = properties.get("model_id") or properties.get(
             "model_dir")
-        device_id = int(properties.get("device_id", "-1"))
-        self.device = f"cuda:{device_id}" if device_id >= 0 else None
-        task = properties.get("task")
-        tp_degree = int(properties.get("tensor_parallel_degree", "-1"))
-        if "trust_remote_code" in properties:
-            self.trust_remote_code = properties.get(
-                "trust_remote_code").lower() == "true"
-        # HF Acc handling
-        kwargs = {"trust_remote_code": self.trust_remote_code}
-        # https://huggingface.co/docs/accelerate/usage_guides/big_modeling#designing-a-device-map
-        if "device_map" in properties:
-            kwargs["device_map"] = properties.get("device_map")
-            self.device = None
-            logging.info(f"Using device map {kwargs['device_map']}")
-        elif tp_degree > 0 and torch.cuda.device_count() > 0:
-            kwargs["device_map"] = "auto"
-            self.device = None
-            world_size = torch.cuda.device_count()
-            assert world_size == tp_degree, f"TP degree ({tp_degree}) doesn't match available GPUs ({world_size})"
-            logging.info(f"Using {world_size} gpus")
-
-        if "data_type" in properties:
-            kwargs["torch_dtype"] = get_torch_dtype_from_str(
-                properties.get("data_type"))
-        if "dtype" in properties:
-            kwargs["torch_dtype"] = get_torch_dtype_from_str(
-                properties.get("dtype"))
-        if "revision" in properties:
-            kwargs["revision"] = properties.get('revision')
-        self.rolling_batch_type = properties.get("rolling_batch", None)
-
-        self._read_model_config(model_id_or_path,
-                                properties.get('revision', None))
+        self.rolling_batch_type = properties.get("rolling_batch", "trtllm").lower()
+        if "trtllm" != self.rolling_batch_type:
+            raise ValueError("only trtllm rollingbatch type supported")
+
+        kwargs = {}
 
         if "output_formatter" in properties:
             kwargs["output_formatter"] = properties.get("output_formatter")
         if "waiting_steps" in properties:
             kwargs["waiting_steps"] = int(properties.get("waiting_steps"))
-        self.rolling_batch_type = self.rolling_batch_type.lower()
-        is_mpi = properties.get("engine") != "Python"
-        if is_mpi:
-            self.device = int(os.getenv("LOCAL_RANK", 0))
-        _rolling_batch_cls = get_rolling_batch_class_from_str(
-            self.rolling_batch_type, is_mpi, self.model_config)
-        self.rolling_batch = _rolling_batch_cls(model_id_or_path, self.device,
+        if "trt_llm_model_name" in properties:
```

Comment:
instead of converting `properties` to `kwargs` in the handler, it's better if we pass the properties to the toolkit and handler everything there. This conversion misses `trt_*` properties that users can set

Reply:
addressed